
Status: I'll merge when I'm added to gcccvs in alioth (rmh).

diff -ur gcc-3.4-3.4.1.old/debian/control.m4 gcc-3.4-3.4.1/debian/control.m4
--- gcc-3.4-3.4.1.old/debian/control.m4	2004-07-23 01:15:50.000000000 +0200
+++ gcc-3.4-3.4.1/debian/control.m4	2004-07-23 01:15:02.000000000 +0200
@@ -37,9 +37,9 @@
 Uploaders: Matthias Klose <doko@debian.org>, Gerhard Tonn <gt@debian.org>
 Standards-Version: 3.6.1
 ifdef(`TARGET',`dnl cross
-Build-Depends: LIBC_BUILD_DEP, m4, autoconf2.13, automake1.7, libtool, autotools-dev, gawk, dpkg-cross (>= 1.14.5), binutils`'TS
+Build-Depends: LIBC_BUILD_DEP, m4, autoconf2.13, autoconf, automake1.7, libtool, autotools-dev, gawk, dpkg-cross (>= 1.14.5), binutils`'TS
 ',`dnl native
-Build-Depends: LIBC_BUILD_DEP, libc6-dev-sparc64 [sparc], libc6-dev-s390x [s390], m4, autoconf2.13, autoconf, automake1.4, automake1.7, libtool, autotools-dev, gawk, dejagnu (>= 1.4.3) [check_no_archs], expect (>= 5.38.0) [check_no_archs], bzip2, BINUTILS_BUILD_DEP, binutils-hppa64 [hppa], debhelper (>= 4.1), gperf (>= 2.7-3), bison (>= 1:1.875a-1) | bison (<< 1:1.50), flex, gettext, texinfo (>= 4.3), zlib1g-dev, libgc-dev [libgc_no_archs], xlibs-dev, gnat-3.3 [ada_no_archs] | gnat-3.4 [ada_no_archs] | gnat [i386 powerpc sparc], libncurses5-dev [pascal_no_archs], libgmp3-dev, tetex-bin [pascal_no_archs], locales [locale_no_archs !hurd-i386], procps [check_no_archs], help2man [pascal_no_archs], sharutils, libgtk2.0-dev [java_no_archs], libart-2.0-dev [java_no_archs], g++-3.3, g77-3.3, gobjc-3.3
+Build-Depends: LIBC_BUILD_DEP, libc6-dev-sparc64 | not+sparc, libc6-dev-s390x | not+s390, m4, autoconf2.13, autoconf, automake1.4, automake1.7, libtool, autotools-dev, gawk, dejagnu (>= 1.4.3) check_no_archs, expect (>= 5.38.0) check_no_archs, bzip2, BINUTILS_BUILD_DEP, binutils-hppa64 | not+hppa, debhelper (>= 4.1), gperf (>= 2.7-3), bison (>= 1:1.875a-1) | bison (<< 1:1.50), flex, gettext, texinfo (>= 4.3), zlib1g-dev, libgc-dev libgc_no_archs, xlibs-dev, gnat-3.3 | gnat-3.4 ada_no_archs | gnat [i386 powerpc sparc], libncurses5-dev pascal_no_archs, libgmp3-dev, tetex-bin pascal_no_archs, locales locale_no_archs, procps check_no_archs, help2man pascal_no_archs, sharutils, libgtk2.0-dev java_no_archs, libart-2.0-dev java_no_archs, g++-3.3, g77-3.3, gobjc-3.3
 Build-Depends-Indep: doxygen (>= 1.3.7)
 ')dnl
 
diff -ur gcc-3.4-3.4.1.old/debian/patches/autoreconf.dpatch gcc-3.4-3.4.1/debian/patches/autoreconf.dpatch
--- gcc-3.4-3.4.1.old/debian/patches/autoreconf.dpatch	2004-07-23 01:15:46.000000000 +0200
+++ gcc-3.4-3.4.1/debian/patches/autoreconf.dpatch	2004-07-23 15:01:32.000000000 +0200
@@ -1,6 +1,6 @@
 #! /bin/sh -e
 
-# DP: autoreconf several directories for proper mipsen libtool support
+# DP: autoreconf several directories for proper k*bsd-gnu libtool support
 
 dir=
 if [ $# -eq 3 -a "$2" = '-d' ]; then
@@ -12,8 +12,11 @@
 fi
 case "$1" in
     -patch)
-	for i in libf2c libjava/libltdl libobjc libstdc++-v3 zlib; do
-		(cd ${dir}/${i} ; autoreconf --force)
+	for i in libf2c libjava libobjc zlib boehm-gc libffi ; do
+		(set -x ; cd ${dir}/${i} ; autoconf2.13 || exit 1)
+	done
+	for i in libstdc++-v3 ; do
+		(set -x ; cd ${dir}/${i} ; autoconf2.50 || exit 1)
 	done
         ;;
     -unpatch)
diff -ur gcc-3.4-3.4.1.old/debian/patches/kbsd-gnu.dpatch gcc-3.4-3.4.1/debian/patches/kbsd-gnu.dpatch
--- gcc-3.4-3.4.1.old/debian/patches/kbsd-gnu.dpatch	2004-07-23 01:15:45.000000000 +0200
+++ gcc-3.4-3.4.1/debian/patches/kbsd-gnu.dpatch	2004-07-23 15:01:43.000000000 +0200
@@ -15,9 +15,6 @@
 case "$1" in
     -patch)
         patch $pdir -f --no-backup-if-mismatch -p1 < $0
-	for i in libffi libf2c libjava/libltdl libobjc libstdc++-v3 zlib; do
-		(set -x ; cd ${dir}/${i} ; autoreconf --force)
-	done
         ;;
     -unpatch)
         patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
@@ -30,9 +27,318 @@
 
 # append the patch here and adjust the -p? flag in the patch calls.
 
-diff -Nur src.old/gcc/config/i386/kfreebsd-gnu.h src/gcc/config/i386/kfreebsd-gnu.h
---- src.old/gcc/config/i386/kfreebsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
-+++ src/gcc/config/i386/kfreebsd-gnu.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.1.old/boehm-gc/configure.in gcc-3.4.1/boehm-gc/configure.in
+--- gcc-3.4.1.old/boehm-gc/configure.in	2004-01-20 16:15:48.000000000 +0100
++++ gcc-3.4.1/boehm-gc/configure.in	2004-07-22 03:56:23.000000000 +0200
+@@ -86,6 +86,9 @@
+ 	AC_DEFINE(GC_LINUX_THREADS)
+ 	AC_DEFINE(_REENTRANT)
+ 	;;
++     *-*-gnu* | *-*-k*bsd*-gnu)
++	AC_DEFINE(_REENTRANT)
++	;;
+      *-*-aix*)
+ 	AC_DEFINE(GC_AIX_THREADS)
+ 	AC_DEFINE(_REENTRANT)
+diff -Nur gcc-3.4.1.old/boehm-gc/dbg_mlc.c gcc-3.4.1/boehm-gc/dbg_mlc.c
+--- gcc-3.4.1.old/boehm-gc/dbg_mlc.c	2003-07-28 06:18:20.000000000 +0200
++++ gcc-3.4.1/boehm-gc/dbg_mlc.c	2004-07-22 03:56:23.000000000 +0200
+@@ -59,7 +59,7 @@
+ 
+ # include <stdlib.h>
+ 
+-# if defined(LINUX) || defined(SUNOS4) || defined(SUNOS5) \
++# if defined(LINUX) || defined(GLIBC) || defined(SUNOS4) || defined(SUNOS5) \
+      || defined(HPUX) || defined(IRIX5) || defined(OSF1)
+ #   define RANDOM() random()
+ # else
+diff -Nur gcc-3.4.1.old/boehm-gc/dyn_load.c gcc-3.4.1/boehm-gc/dyn_load.c
+--- gcc-3.4.1.old/boehm-gc/dyn_load.c	2003-07-30 19:42:28.000000000 +0200
++++ gcc-3.4.1/boehm-gc/dyn_load.c	2004-07-22 03:56:23.000000000 +0200
+@@ -26,7 +26,8 @@
+  * None of this is safe with dlclose and incremental collection.
+  * But then not much of anything is safe in the presence of dlclose.
+  */
+-#if defined(__linux__) && !defined(_GNU_SOURCE)
++#if (defined(__linux__) || defined(__GNU__) || defined(__GLIBC__)) \
++  && !defined(_GNU_SOURCE)
+     /* Can't test LINUX, since this must be define before other includes */
+ #   define _GNU_SOURCE
+ #endif
+@@ -56,7 +57,7 @@
+     !(defined(ALPHA) && defined(OSF1)) && \
+     !defined(HPUX) && !(defined(LINUX) && defined(__ELF__)) && \
+     !defined(RS6000) && !defined(SCO_ELF) && !defined(DGUX) && \
+-    !(defined(FREEBSD) && defined(__ELF__)) && \
++    !(defined(KFREEBSD) && defined(__ELF__)) && \
+     !(defined(NETBSD) && defined(__ELF__)) && !defined(HURD) && \
+     !defined(DARWIN)
+  --> We only know how to find data segments of dynamic libraries for the
+@@ -81,7 +82,7 @@
+ #endif
+ 
+ #if defined(LINUX) && defined(__ELF__) || defined(SCO_ELF) || \
+-    (defined(FREEBSD) && defined(__ELF__)) || defined(DGUX) || \
++    (defined(KFREEBSD) && defined(__ELF__)) || defined(DGUX) || \
+     (defined(NETBSD) && defined(__ELF__)) || defined(HURD)
+ #   include <stddef.h>
+ #   include <elf.h>
+@@ -265,7 +266,7 @@
+ # endif /* SUNOS */
+ 
+ #if defined(LINUX) && defined(__ELF__) || defined(SCO_ELF) || \
+-    (defined(FREEBSD) && defined(__ELF__)) || defined(DGUX) || \
++    (defined(KFREEBSD) && defined(__ELF__)) || defined(DGUX) || \
+     (defined(NETBSD) && defined(__ELF__)) || defined(HURD)
+ 
+ 
+@@ -360,7 +361,7 @@
+ /* For glibc 2.2.4+.  Unfortunately, it doesn't work for older	*/
+ /* versions.  Thanks to Jakub Jelinek for most of the code.	*/
+ 
+-# if defined(LINUX) /* Are others OK here, too? */ \
++# if defined(GLIBC) \
+      && (__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ > 2) \
+          || (__GLIBC__ == 2 && __GLIBC_MINOR__ == 2 && defined(DT_CONFIG))) 
+ 
+@@ -434,7 +435,7 @@
+ 
+ #define HAVE_REGISTER_MAIN_STATIC_DATA
+ 
+-# else /* !LINUX || version(glibc) < 2.2.4 */
++# else /* !GLIBC || version(glibc) < 2.2.4 */
+ 
+ /* Dynamic loading code for Linux running ELF. Somewhat tested on
+  * Linux/x86, untested but hopefully should work on Linux/Alpha. 
+diff -Nur gcc-3.4.1.old/boehm-gc/include/gc.h gcc-3.4.1/boehm-gc/include/gc.h
+--- gcc-3.4.1.old/boehm-gc/include/gc.h	2003-07-31 06:52:36.000000000 +0200
++++ gcc-3.4.1/boehm-gc/include/gc.h	2004-07-22 03:56:23.000000000 +0200
+@@ -466,7 +466,7 @@
+ #   define GC_RETURN_ADDR (GC_word)__return_address
+ #endif
+ 
+-#ifdef __linux__
++#if defined(__linux__) || defined(__GNU__) || defined(__GLIBC__)
+ # include <features.h>
+ # if (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 1 || __GLIBC__ > 2) \
+      && !defined(__ia64__)
+@@ -490,7 +490,8 @@
+ /* This may also be desirable if it is possible but expensive to	*/
+ /* retrieve the call chain.						*/
+ #if (defined(__linux__) || defined(__NetBSD__) || defined(__OpenBSD__) \
+-     || defined(__FreeBSD__)) & !defined(GC_CAN_SAVE_CALL_STACKS)
++     || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) \
++     ) & !defined(GC_CAN_SAVE_CALL_STACKS)
+ # define GC_ADD_CALLER
+ # if __GNUC__ >= 3 || (__GNUC__ == 2 && __GNUC_MINOR__ >= 95) 
+     /* gcc knows how to retrieve return address, but we don't know */
+diff -Nur gcc-3.4.1.old/boehm-gc/include/private/gcconfig.h gcc-3.4.1/boehm-gc/include/private/gcconfig.h
+--- gcc-3.4.1.old/boehm-gc/include/private/gcconfig.h	2004-04-21 17:17:48.000000000 +0200
++++ gcc-3.4.1/boehm-gc/include/private/gcconfig.h	2004-07-22 03:56:24.000000000 +0200
+@@ -44,6 +44,14 @@
+ #  endif
+ # endif
+ 
++/* And one for Glibc: */
++#if defined(LINUX) || defined(__GNU__) || defined(__GLIBC__)
++# include <features.h>
++#endif
++#ifdef __GLIBC__
++# define GLIBC
++#endif
++
+ /* And one for NetBSD: */
+ # if defined(__NetBSD__)
+ #    define NETBSD
+@@ -58,6 +66,9 @@
+ # if defined(__FreeBSD__)
+ #    define FREEBSD
+ # endif
++# if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
++#    define KFREEBSD
++# endif
+ 
+ /* Determine the machine type: */
+ # if defined(__arm__) || defined(__thumb__)
+@@ -299,7 +310,7 @@
+ #   define OPENBSD
+ #   define mach_type_known
+ # endif
+-# if defined(FREEBSD) && (defined(i386) || defined(__i386__))
++# if defined(KFREEBSD) && (defined(i386) || defined(__i386__))
+ #   define I386
+ #   define mach_type_known
+ # endif
+@@ -1162,13 +1173,25 @@
+ #	endif
+ #	define SIG_SUSPEND SIGUSR1
+ #	define SIG_THR_RESTART SIGUSR2
++	extern char etext[];
++	extern char * GC_FreeBSDGetDataStart();
++#	define DATASTART GC_FreeBSDGetDataStart(0x1000, &etext)
++#   endif
++#   if defined(GLIBC) && defined(KFREEBSD)
++#	define OS_TYPE "GNU/kFreeBSD"
++#	if !defined(GC_LINUX_THREADS) && !defined(REDIRECT_MALLOC)
++#	    define MPROTECT_VDB
++#	endif
++#	define SIG_SUSPEND 57 /* arbitrary signal number <= 128 */
++#	define SIG_THR_RESTART 58 /* arbitrary signal number <= 128 */
++#	define SEARCH_FOR_DATA_START
++#	define DATAEND (sbrk (0))
++#   endif
++#   ifdef KFREEBSD
+ #	define FREEBSD_STACKBOTTOM
+ #	ifdef __ELF__
+ #	    define DYNAMIC_LOADING
+ #	endif
+-	extern char etext[];
+-	extern char * GC_FreeBSDGetDataStart();
+-#	define DATASTART GC_FreeBSDGetDataStart(0x1000, &etext)
+ #   endif
+ #   ifdef NETBSD
+ #	define OS_TYPE "NETBSD"
+diff -Nur gcc-3.4.1.old/boehm-gc/os_dep.c gcc-3.4.1/boehm-gc/os_dep.c
+--- gcc-3.4.1.old/boehm-gc/os_dep.c	2003-09-22 18:00:23.000000000 +0200
++++ gcc-3.4.1/boehm-gc/os_dep.c	2004-07-22 03:56:24.000000000 +0200
+@@ -80,7 +80,7 @@
+ #   define NEED_FIND_LIMIT
+ # endif
+ 
+-#if defined(FREEBSD) && defined(I386)
++#if defined(KFREEBSD) && defined(I386)
+ #  include <machine/trap.h>
+ #  if !defined(PCR)
+ #    define NEED_FIND_LIMIT
+@@ -121,7 +121,7 @@
+ # include <fcntl.h>
+ #endif
+ 
+-#if defined(SUNOS5SIGS) || defined (HURD) || defined(LINUX)
++#if defined(SUNOS5SIGS) || defined(LINUX) || defined(GLIBC)
+ # ifdef SUNOS5SIGS
+ #  include <sys/siginfo.h>
+ # endif
+@@ -311,7 +311,7 @@
+   /* for recent Linux versions.  This seems to be the easiest way to	*/
+   /* cover all versions.						*/
+ 
+-# ifdef LINUX
++# if defined(LINUX) || defined(GLIBC)
+     /* Some Linux distributions arrange to define __data_start.  Some	*/
+     /* define data_start as a weak symbol.  The latter is technically	*/
+     /* broken, since the user program may define data_start, in which	*/
+@@ -321,7 +321,7 @@
+     extern int __data_start[];
+ #   pragma weak data_start
+     extern int data_start[];
+-# endif /* LINUX */
++# endif /* LINUX || GLIBC */
+   extern int _end[];
+ 
+   ptr_t GC_data_start;
+@@ -330,7 +330,7 @@
+   {
+     extern ptr_t GC_find_limit();
+ 
+-#   ifdef LINUX
++#   if defined(LINUX) || defined(GLIBC)
+       /* Try the easy approaches first:	*/
+       if ((ptr_t)__data_start != 0) {
+ 	  GC_data_start = (ptr_t)(__data_start);
+@@ -340,7 +340,7 @@
+ 	  GC_data_start = (ptr_t)(data_start);
+ 	  return;
+       }
+-#   endif /* LINUX */
++#   endif /* LINUX || GLIBC */
+     GC_data_start = GC_find_limit((ptr_t)(_end), FALSE);
+   }
+ #endif
+@@ -2175,13 +2175,13 @@
+ #endif /* SUNOS4 || FREEBSD */
+ 
+ #if defined(SUNOS5SIGS) || defined(OSF1) || defined(LINUX) \
+-    || defined(HURD)
++    || defined(GLIBC)
+ # ifdef __STDC__
+     typedef void (* SIG_PF)(int);
+ # else
+     typedef void (* SIG_PF)();
+ # endif
+-#endif /* SUNOS5SIGS || OSF1 || LINUX || HURD */
++#endif /* SUNOS5SIGS || OSF1 || LINUX || GLIBC */
+ 
+ #if defined(MSWIN32)
+     typedef LPTOP_LEVEL_EXCEPTION_FILTER SIG_PF;
+@@ -2301,7 +2301,7 @@
+ 
+ /*ARGSUSED*/
+ #if !defined(DARWIN)
+-# if defined (SUNOS4) || defined(FREEBSD)
++# if defined (SUNOS4) || defined(KFREEBSD)
+     void GC_write_fault_handler(sig, code, scp, addr)
+     int sig, code;
+     struct sigcontext *scp;
+@@ -2312,11 +2312,11 @@
+               	    || (FC_CODE(code) == FC_OBJERR \
+               	       && FC_ERRNO(code) == FC_PROT))
+ #   endif
+-#   ifdef FREEBSD
++#   ifdef KFREEBSD
+ #     define SIG_OK (sig == SIGBUS)
+ #     define CODE_OK (code == BUS_PAGE_FAULT)
+ #   endif
+-# endif /* SUNOS4 || FREEBSD */
++# endif /* SUNOS4 || KFREEBSD */
+ 
+ # if defined(IRIX5) || defined(OSF1) || defined(HURD)
+ #   include <errno.h>
+@@ -3833,7 +3833,7 @@
+ /* I suspect the following works for most X86 *nix variants, so 	*/
+ /* long as the frame pointer is explicitly stored.  In the case of gcc,	*/
+ /* compiler flags (e.g. -fomit-frame-pointer) determine whether it is.	*/
+-#if defined(I386) && defined(LINUX) && defined(SAVE_CALL_CHAIN)
++#if defined(I386) && (defined(LINUX) || defined(GLIBC)) && defined(SAVE_CALL_CHAIN)
+ #   include <features.h>
+ 
+     struct frame {
+@@ -3844,7 +3844,7 @@
+ #endif
+ 
+ #if defined(SPARC)
+-#  if defined(LINUX)
++#  if defined(LINUX) || defined(GLIBC)
+ #    include <features.h>
+ 
+      struct frame {
+diff -Nur gcc-3.4.1.old/configure gcc-3.4.1/configure
+--- gcc-3.4.1.old/configure	2004-06-25 23:33:15.000000000 +0200
++++ gcc-3.4.1/configure	2004-07-22 03:56:24.000000000 +0200
+@@ -1928,7 +1928,7 @@
+   powerpc-*-netware*)
+     target_makefile_frag="config/mt-netware"
+     ;;
+-  *-*-linux*)
++  *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu)
+     target_makefile_frag="config/mt-linux"
+     ;;
+   *-*-aix4.[3456789]* | *-*-aix[56789].*)
+diff -Nur gcc-3.4.1.old/configure.in gcc-3.4.1/configure.in
+--- gcc-3.4.1.old/configure.in	2004-06-25 23:33:14.000000000 +0200
++++ gcc-3.4.1/configure.in	2004-07-22 03:56:24.000000000 +0200
+@@ -1167,7 +1167,7 @@
+   powerpc-*-netware*)
+     target_makefile_frag="config/mt-netware"
+     ;;
+-  *-*-linux*)
++  *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu)
+     target_makefile_frag="config/mt-linux"
+     ;;
+   *-*-aix4.[[3456789]]* | *-*-aix[[56789]].*)
+diff -Nur gcc-3.4.1.old/gcc/config/i386/kfreebsd-gnu.h gcc-3.4.1/gcc/config/i386/kfreebsd-gnu.h
+--- gcc-3.4.1.old/gcc/config/i386/kfreebsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
++++ gcc-3.4.1/gcc/config/i386/kfreebsd-gnu.h	2004-07-22 03:56:25.000000000 +0200
 @@ -0,0 +1,26 @@
 +/* Definitions for Intel 386 running kFreeBSD-based GNU systems with ELF format
 +   Copyright (C) 2004
@@ -60,9 +366,9 @@
 +#define LINK_EMULATION "elf_i386_fbsd"
 +#undef REG_NAME
 +#define REG_NAME(reg) sc_ ## reg
-diff -Nur src.old/gcc/config/i386/knetbsd-gnu.h src/gcc/config/i386/knetbsd-gnu.h
---- src.old/gcc/config/i386/knetbsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
-+++ src/gcc/config/i386/knetbsd-gnu.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.1.old/gcc/config/i386/knetbsd-gnu.h gcc-3.4.1/gcc/config/i386/knetbsd-gnu.h
+--- gcc-3.4.1.old/gcc/config/i386/knetbsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
++++ gcc-3.4.1/gcc/config/i386/knetbsd-gnu.h	2004-07-22 03:56:25.000000000 +0200
 @@ -0,0 +1,24 @@
 +/* Definitions for Intel 386 running kNetBSD-based GNU systems with ELF format
 +   Copyright (C) 2004
@@ -88,9 +394,9 @@
 +
 +#undef REG_NAME
 +#define REG_NAME(reg) sc_ ## reg
-diff -Nur src.old/gcc/config/i386/linux.h src/gcc/config/i386/linux.h
---- src.old/gcc/config/i386/linux.h	2003-11-29 04:08:10.000000000 +0100
-+++ src/gcc/config/i386/linux.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.1.old/gcc/config/i386/linux.h gcc-3.4.1/gcc/config/i386/linux.h
+--- gcc-3.4.1.old/gcc/config/i386/linux.h	2003-11-29 04:08:10.000000000 +0100
++++ gcc-3.4.1/gcc/config/i386/linux.h	2004-07-22 03:56:25.000000000 +0200
 @@ -108,24 +108,30 @@
  
  /* If ELF is the default format, we should not use /lib/elf.  */
@@ -181,9 +487,9 @@
      (FS)->retaddr_column = 8;						\
      goto SUCCESS;							\
    } while (0)
-diff -Nur src.old/gcc/config/kfreebsd-gnu.h src/gcc/config/kfreebsd-gnu.h
---- src.old/gcc/config/kfreebsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
-+++ src/gcc/config/kfreebsd-gnu.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.1.old/gcc/config/kfreebsd-gnu.h gcc-3.4.1/gcc/config/kfreebsd-gnu.h
+--- gcc-3.4.1.old/gcc/config/kfreebsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
++++ gcc-3.4.1/gcc/config/kfreebsd-gnu.h	2004-07-22 03:56:25.000000000 +0200
 @@ -0,0 +1,36 @@
 +/* Definitions for kFreeBSD-based GNU systems with ELF format
 +   Copyright (C) 2004
@@ -221,9 +527,9 @@
 +
 +#undef DYNAMIC_LINKER
 +#define DYNAMIC_LINKER "/lib/ld.so.1"
-diff -Nur src.old/gcc/config/knetbsd-gnu.h src/gcc/config/knetbsd-gnu.h
---- src.old/gcc/config/knetbsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
-+++ src/gcc/config/knetbsd-gnu.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.1.old/gcc/config/knetbsd-gnu.h gcc-3.4.1/gcc/config/knetbsd-gnu.h
+--- gcc-3.4.1.old/gcc/config/knetbsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
++++ gcc-3.4.1/gcc/config/knetbsd-gnu.h	2004-07-22 03:56:25.000000000 +0200
 @@ -0,0 +1,36 @@
 +/* Definitions for kNetBSD-based GNU systems with ELF format
 +   Copyright (C) 2004
@@ -261,9 +567,9 @@
 +
 +#undef DYNAMIC_LINKER
 +#define DYNAMIC_LINKER "/lib/ld.so.1"
-diff -Nur src.old/gcc/config/linux.h src/gcc/config/linux.h
---- src.old/gcc/config/linux.h	2003-11-29 04:08:10.000000000 +0100
-+++ src/gcc/config/linux.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.1.old/gcc/config/linux.h gcc-3.4.1/gcc/config/linux.h
+--- gcc-3.4.1.old/gcc/config/linux.h	2003-11-29 04:08:10.000000000 +0100
++++ gcc-3.4.1/gcc/config/linux.h	2004-07-22 03:56:25.000000000 +0200
 @@ -97,6 +97,7 @@
         %{!p:%{!pg:%{!g*:-lc} %{g*:-lg}}}}"
  #endif
@@ -280,9 +586,9 @@
  
  #if !defined(USE_GNULIBC_1) && defined(HAVE_LD_EH_FRAME_HDR)
  #define LINK_EH_SPEC "%{!static:--eh-frame-hdr} "
-diff -Nur src.old/gcc/config.gcc src/gcc/config.gcc
---- src.old/gcc/config.gcc	2004-05-22 01:09:22.000000000 +0200
-+++ src/gcc/config.gcc	2004-05-22 02:41:54.000000000 +0200
+diff -Nur gcc-3.4.1.old/gcc/config.gcc gcc-3.4.1/gcc/config.gcc
+--- gcc-3.4.1.old/gcc/config.gcc	2004-04-21 17:12:35.000000000 +0200
++++ gcc-3.4.1/gcc/config.gcc	2004-07-22 03:56:25.000000000 +0200
 @@ -428,21 +428,10 @@
    esac
    fbsd_tm_file="${fbsd_tm_file} freebsd-spec.h freebsd.h"
@@ -336,9 +642,43 @@
  i[34567]86-*-gnu*)
  	;;
  i[34567]86-pc-msdosdjgpp*)
-diff -Nur src.old/libtool.m4 src/libtool.m4
---- src.old/libtool.m4	2003-11-19 06:29:32.000000000 +0100
-+++ src/libtool.m4	2004-05-22 02:37:46.000000000 +0200
+diff -Nur gcc-3.4.1.old/libffi/configure.in gcc-3.4.1/libffi/configure.in
+--- gcc-3.4.1.old/libffi/configure.in	2004-04-27 07:10:19.000000000 +0200
++++ gcc-3.4.1/libffi/configure.in	2004-07-22 03:56:26.000000000 +0200
+@@ -60,16 +60,16 @@
+ i*86-*-linux*) TARGET=X86; TARGETDIR=x86;;
+ i*86-*-solaris*) TARGET=X86; TARGETDIR=x86;;
+ i*86-*-beos*) TARGET=X86; TARGETDIR=x86;;
+-i*86-*-freebsd*) TARGET=X86; TARGETDIR=x86;;
+-i*86-*-netbsdelf*) TARGET=X86; TARGETDIR=x86;;
++i*86-*-freebsd* | i*86-*-kfreebsd*-gnu) TARGET=X86; TARGETDIR=x86;;
++i*86-*-netbsdelf* | i*86-*-knetbsd*-gnu) TARGET=X86; TARGETDIR=x86;;
+ i*86-*-win32*) TARGET=X86_WIN32; TARGETDIR=x86;;
+ i*86-*-cygwin*) TARGET=X86_WIN32; TARGETDIR=x86;;
+ i*86-*-mingw*) TARGET=X86_WIN32; TARGETDIR=x86;;
+ sparc-sun-4*) TARGET=SPARC; TARGETDIR=sparc;;
+ sparc*-sun-*) TARGET=SPARC; TARGETDIR=sparc;;
+-sparc-*-linux* | sparc-*-netbsdelf*) TARGET=SPARC; TARGETDIR=sparc;;
+-sparc64-*-linux* | sparc64-*-netbsd*) TARGET=SPARC; TARGETDIR=sparc;;
+-alpha*-*-linux* | alpha*-*-osf* | alpha*-*-freebsd* | alpha*-*-netbsd*) TARGET=ALPHA; TARGETDIR=alpha;;
++sparc-*-linux* | sparc-*-netbsdelf* | sparc-*-knetbsd*-gnu) TARGET=SPARC; TARGETDIR=sparc;;
++sparc64-*-linux* | sparc64-*-netbsd* | sparc64-*-knetbsd*-gnu) TARGET=SPARC; TARGETDIR=sparc;;
++alpha*-*-linux* | alpha*-*-osf* | alpha*-*-freebsd* | alpha*-*-kfreebsd*-gnu | alpha*-*-netbsd* | alpha*-*-knetbsd*-gnu) TARGET=ALPHA; TARGETDIR=alpha;;
+ ia64*-*-*) TARGET=IA64; TARGETDIR=ia64;;
+ m68k-*-linux*) TARGET=M68K; TARGETDIR=m68k;;
+ mips64*-*);;
+@@ -81,7 +81,7 @@
+ powerpc-*-aix*) TARGET=POWERPC_AIX; TARGETDIR=powerpc;;
+ rs6000-*-aix*) TARGET=POWERPC_AIX; TARGETDIR=powerpc;;
+ arm*-*-linux-*) TARGET=ARM; TARGETDIR=arm;;
+-arm*-*-netbsdelf*) TARGET=ARM; TARGETDIR=arm;;
++arm*-*-netbsdelf* | i*86-*-knetbsd*-gnu) TARGET=ARM; TARGETDIR=arm;;
+ s390-*-linux-*) TARGET=S390; TARGETDIR=s390;;
+ s390x-*-linux-*) TARGET=S390; TARGETDIR=s390;;
+ x86_64-*-linux*) TARGET=X86_64; TARGETDIR=x86;;
+diff -Nur gcc-3.4.1.old/libtool.m4 gcc-3.4.1/libtool.m4
+--- gcc-3.4.1.old/libtool.m4	2004-05-18 11:08:37.000000000 +0200
++++ gcc-3.4.1/libtool.m4	2004-07-22 03:56:26.000000000 +0200
 @@ -621,7 +621,7 @@
    lt_cv_deplibs_check_method=pass_all
    ;;
@@ -357,9 +697,9 @@
    if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
      [lt_cv_deplibs_check_method='match_pattern /lib[^/\.]+\.so\.[0-9]+\.[0-9]+$']
    else
-diff -Nur src.old/ltcf-c.sh src/ltcf-c.sh
---- src.old/ltcf-c.sh	2003-11-19 06:29:32.000000000 +0100
-+++ src/ltcf-c.sh	2004-05-22 02:37:46.000000000 +0200
+diff -Nur gcc-3.4.1.old/ltcf-c.sh gcc-3.4.1/ltcf-c.sh
+--- gcc-3.4.1.old/ltcf-c.sh	2003-11-19 06:29:32.000000000 +0100
++++ gcc-3.4.1/ltcf-c.sh	2004-07-22 03:56:26.000000000 +0200
 @@ -175,7 +175,7 @@
        $CC $output_objdir/$soname-exp '$lt_cv_cc_dll_switch' -Wl,-e,'$dll_entry' -o $output_objdir/$soname '$ltdll_obj'$libobjs $deplibs $compiler_flags'
      ;;
@@ -387,9 +727,9 @@
      if echo __ELF__ | $CC -E - | grep __ELF__ >/dev/null; then
        archive_cmds='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags'  # a.out
      else
-diff -Nur src.old/ltcf-cxx.sh src/ltcf-cxx.sh
---- src.old/ltcf-cxx.sh	2003-11-19 06:29:32.000000000 +0100
-+++ src/ltcf-cxx.sh	2004-05-22 02:37:46.000000000 +0200
+diff -Nur gcc-3.4.1.old/ltcf-cxx.sh gcc-3.4.1/ltcf-cxx.sh
+--- gcc-3.4.1.old/ltcf-cxx.sh	2003-11-19 06:29:32.000000000 +0100
++++ gcc-3.4.1/ltcf-cxx.sh	2004-07-22 03:56:26.000000000 +0200
 @@ -289,7 +289,7 @@
      # C++ shared libraries reported to be fairly broken before switch to ELF
      ld_shlibs=no
@@ -417,9 +757,9 @@
        # FreeBSD uses GNU C++
        ;;
      gnu*)
-diff -Nur src.old/ltcf-gcj.sh src/ltcf-gcj.sh
---- src.old/ltcf-gcj.sh	2003-11-19 06:29:32.000000000 +0100
-+++ src/ltcf-gcj.sh	2004-05-22 02:37:46.000000000 +0200
+diff -Nur gcc-3.4.1.old/ltcf-gcj.sh gcc-3.4.1/ltcf-gcj.sh
+--- gcc-3.4.1.old/ltcf-gcj.sh	2003-11-19 06:29:32.000000000 +0100
++++ gcc-3.4.1/ltcf-gcj.sh	2004-07-22 03:56:26.000000000 +0200
 @@ -178,7 +178,7 @@
        $CC $output_objdir/$soname-exp '$lt_cv_cc_dll_switch' -Wl,-e,'$dll_entry' -o $output_objdir/$soname '$ltdll_obj'$libobjs $deplibs $compiler_flags'
      ;;
@@ -447,9 +787,9 @@
      if echo __ELF__ | $CC -E - | grep __ELF__ >/dev/null; then
        archive_cmds='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags'  # a.out
      else
-diff -Nur src.old/ltconfig src/ltconfig
---- src.old/ltconfig	2004-03-05 22:05:41.000000000 +0100
-+++ src/ltconfig	2004-05-22 02:37:46.000000000 +0200
+diff -Nur gcc-3.4.1.old/ltconfig gcc-3.4.1/ltconfig
+--- gcc-3.4.1.old/ltconfig	2004-03-05 22:05:41.000000000 +0100
++++ gcc-3.4.1/ltconfig	2004-07-22 03:56:26.000000000 +0200
 @@ -1168,6 +1168,17 @@
    hardcode_into_libs=yes
    ;;
diff -ur gcc-3.4-3.4.1.old/debian/rules.conf gcc-3.4-3.4.1/debian/rules.conf
--- gcc-3.4-3.4.1.old/debian/rules.conf	2004-07-23 01:15:50.000000000 +0200
+++ gcc-3.4-3.4.1/debian/rules.conf	2004-07-23 01:15:02.000000000 +0200
@@ -171,12 +171,12 @@
 	  -DGNAT_V=$(GNAT_VERSION) \
 	  -DFFI_SO=$(FFI_SONAME) \
 	  -Denabled_languages="$(languages) $(addons)" \
-	  -Dada_no_archs="$(foreach arch,$(ada_no_cpus) $(ada_no_systems),!$(arch))" \
-	  -Djava_no_archs="$(foreach arch,$(java_no_cpus) $(java_no_systems),!$(arch))" \
-	  -Dpascal_no_archs="$(foreach arch,$(pascal_no_cpus) $(pascal_no_systems),!$(arch))" \
-	  -Dlibgc_no_archs="$(foreach arch,$(libgc_no_cpus) $(libgc_no_systems),!$(arch))" \
-	  -Dcheck_no_archs="$(foreach arch,$(check_no_cpus) $(check_no_systems),!$(arch))" \
-	  -Dlocale_no_archs="$(foreach arch,$(locale_no_cpus) $(locale_no_systems),!$(arch))" \
+	  -Dada_no_archs="$(foreach arch,$(ada_no_cpus) $(ada_no_systems), | $(arch))" \
+	  -Djava_no_archs="$(foreach arch,$(java_no_cpus) $(java_no_systems), | $(arch))" \
+	  -Dpascal_no_archs="$(foreach arch,$(pascal_no_cpus) $(pascal_no_systems), | $(arch))" \
+	  -Dlibgc_no_archs="$(foreach arch,$(libgc_no_cpus) $(libgc_no_systems), | $(arch))" \
+	  -Dcheck_no_archs="$(foreach arch,$(check_no_cpus) $(check_no_systems), | $(arch))" \
+	  -Dlocale_no_archs="$(foreach arch,$(locale_no_cpus) $(locale_no_systems), | $(arch))" \
 		debian/control.m4 > debian/control.tmp2
 	uniq debian/control.tmp2 > debian/control.tmp
 	rm -f debian/control.tmp2
diff -ur gcc-3.4-3.4.1.old/debian/rules.defs gcc-3.4-3.4.1/debian/rules.defs
--- gcc-3.4-3.4.1.old/debian/rules.defs	2004-07-23 01:15:50.000000000 +0200
+++ gcc-3.4-3.4.1/debian/rules.defs	2004-07-23 03:57:00.000000000 +0200
@@ -192,7 +192,7 @@
   with_java := yes
 endif
 
-java_no_systems := gnu kfreebsd-gnu knetbsd-gnu netbsd-elf-gnu
+java_no_systems := gnu knetbsd-gnu netbsd-elf-gnu
 ifeq ($(DEB_TARGET_GNU_SYSTEM), $(findstring $(DEB_TARGET_GNU_SYSTEM),$(java_no_systems)))
   with_java := disabled for $(DEB_TARGET_GNU_SYSTEM)
 endif
@@ -239,7 +239,7 @@
 #ifeq ($(with_common_libs),yes)
   with_libffi := yes
   no_ffi_cpus :=
-  no_ffi_systems := gnu kfreebsd-gnu knetbsd-gnu netbsd-elf-gnu
+  no_ffi_systems := gnu knetbsd-gnu netbsd-elf-gnu
   ifneq ($(with_java),yes)
     ifeq ($(DEB_TARGET_GNU_CPU), $(findstring $(DEB_TARGET_GNU_CPU),$(no_ffi_cpus)))
       with_libffi := disabled for architecure $(DEB_TARGET_GNU_CPU)
@@ -347,7 +347,7 @@
 
 # disable ObjC garbage collection library (needs libgc)
 libgc_no_cpus := avr
-libgc_no_systems := kfreebsd-gnu knetbsd-gnu
+libgc_no_systems := knetbsd-gnu
 ifeq ($(DEB_TARGET_GNU_CPU),$(findstring $(DEB_TARGET_GNU_CPU),$(libgc_no_cpus)))
   with_objc_gc := disabled for architecture $(DEB_TARGET_GNU_CPU)
 endif
@@ -465,7 +465,7 @@
 ifdef DEB_CROSS
   with_check := disabled for cross compiler package
 endif
-check_no_systems := gnu kfreebsd-gnu knetbsd-gnu
+check_no_systems := gnu
 ifeq ($(DEB_TARGET_GNU_SYSTEM), $(findstring $(DEB_TARGET_GNU_SYSTEM),$(check_no_systems)))
   with_check := disabled for $(DEB_TARGET_GNU_SYSTEM)
 endif
diff -ur gcc-3.4-3.4.1.old/debian/rules.patch gcc-3.4-3.4.1/debian/rules.patch
--- gcc-3.4-3.4.1.old/debian/rules.patch	2004-07-23 01:15:50.000000000 +0200
+++ gcc-3.4-3.4.1/debian/rules.patch	2004-07-23 15:02:20.000000000 +0200
@@ -119,6 +119,8 @@
 
 debian_patches += link-libs reporting
 
+# always be last
+debian_patches += autoreconf
 
 patch: $(patch_stamp)
 $(patch_stamp): $(unpack_stamp) pre-patch \
